metrics/recorder_test.go

```diff
@@ -180,11 +185,22 @@ func TestRecorder_ConcurrentSafety(t *testing.T) {
 	go thread()
 	go thread()
 
-	<-ch
-	<-ch
+	for finishedThreads := 0; finishedThreads < 2; {
+		select {
+		case <-ctx.Done():
```

Comment:
is there a point in using context here if you're not passing it around? why not just use <-time.After(...) instead?

Reply:
Didn't know about that one, thanks!